app/models/team.rb

```diff
@@ -587,6 +595,31 @@ def get_feed(feed_id)
     self.feeds.where(id: feed_id.to_i).last
   end
 
+  # A newsletter header type is available only if there are WhatsApp templates for it
+  def available_newsletter_header_types
+    header_type_mapping = {
+      'none' => 'none',
+      'image' => 'image',
+      'video' => 'video',
+      'audio' => 'video', # WhatsApp doesn't support audio header, so we convert it to video
+      'link_preview' => 'none'
+    }
+    available = []
+    tbi = TeamBotInstallation.where(team_id: self.id, user_id: BotUser.smooch_user&.id.to_i).last
+    unless tbi.nil?
+      ['none', 'image', 'video', 'audio', 'link_preview'].each do |header_type|
+        mapped_header_type = header_type_mapping[header_type]
+        if !tbi.send("get_smooch_template_name_for_newsletter_#{mapped_header_type}_no_articles").blank? &&
```

Comment:
Yes, that's a good point! And I think we should revisit that in many places of the application, because it's confusing and makes it hard to debug. In this case here, these methods are just shortcuts for settings keys... so, `object.get_foo_bar` is the same as `object.settings['foo_bar']`. I could replace by the latter for simplicity?